src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -171,6 +171,42 @@ def get_component_contexts(
                 output.append(context_item)
         return output
 
+    def get_installation_component_contexts(
+        self,
+        *,
+        filter: SentryAppInstallationFilterArgs,
+        component_type: str,
+        include_contexts_with_component: bool = True,
+    ) -> list[RpcSentryAppComponentContext]:
+        install_query = self._FQ.query_many(filter=filter)
+        app_ids = install_query.select_related("sentry_app", "sentry_app__application").values_list(
+            "sentry_app_id", flat=True
+        )
+        component_query = SentryAppComponent.objects.filter(
+            type=component_type, sentry_app_id__in=list(app_ids)
+        )
+        component_map: dict[int, SentryAppComponent] = {}
+        output = []
+
+        for component in component_query:
+            component_map[component.sentry_app.id] = component
+
+        for install in install_query:
+            install_component = component_map.get(install.sentry_app_id)
+            if include_contexts_with_component or install_component:
+                context_item = RpcSentryAppComponentContext(
+                    installation=serialize_sentry_app_installation(
+                        installation=install, app=install.sentry_app
```

Comment:
Do we know how many sentry apps a single query will span?

Reply:
I don't think I understand the question?

Reply:
technically this RPC method could fetch all published/unpublished sentry apps which could be a lot, but the current uses of the `get_component_contexts` RPC method filter by `uuid`, `app_id`, or `organization_id` which would return very few sentry apps

the method requires some sort of filtering (see filter args below)
https://github.com/getsentry/sentry/blob/e277c9aeba9968ed5ba1347b61ef8b1e78ed7293/src/sentry/sentry_apps/services/app/model.py#L150-L157

Reply:
Discussed in person, makes sense that it could be multiple. Consolidating these into single queries per entity could help with the n+1 queries we're potentially doing here.

Reply:
Is this referring to the uses of `install.sentry_app` and `install.sentry_app_id`?

Reply:
Would updating the initial `install_query` to use `.select_related()` solve the problem?

so updating this
https://github.com/getsentry/sentry/blob/ed9f234cb11f49fc855c78808009e028d6c3eaee/src/sentry/sentry_apps/services/app/impl.py#L181

to be
```
        install_query = self._FQ.query_many(filter=filter).select_related("sentry_app")
```

Reply:
oh yes that should help!

Reply:
Yes, you need a `.select_related()` to avoid the N+1 query, but the default `_AppServiceFilterQuery.base_query()` will include a `select_related()` for you.

Reply:
@markstory ah does this `base_query()` get applied to all queries?